app-backend/datamodel/src/main/scala/com/azavea/rf/datamodel/ColorCorrect.scala

```diff
@@ -153,25 +153,65 @@ object ColorCorrect {
+                          acc && (hst match {
+                            case x if 1 until 255 contains x.minValue().map(_.toInt).getOrElse(0) => true
+                            case x if 1 until 255 contains x.maxValue().map(_.toInt).getOrElse(0) => true
+                            case _ => false
+                          })
+                        )})
+      layerClp      <- if (!isCorrected) {
+                          val getMin = (hst:Histogram[Double]) => hst.minValue().map(_.toInt).getOrElse(0)
+                          val getMax = (hst:Histogram[Double]) => hst.maxValue().map(_.toInt).getOrElse(255)
+                          Some(LayerClipping(getMin(r), getMax(r), getMin(g), getMax(g), getMin(b), getMax(b)))
+                        } else { None }
+    } yield layerClp).getOrElse(LayerClipping(0, 255, 0, 255, 0, 255))
+
     val rgbBand =
       (specificBand:Option[Int], allBands:Option[Int], tileDefault: Int) =>
         specificBand.fold(allBands)(Some(_)).fold(Some(tileDefault))(x => Some(x))
+
+    val layerNormalizeArgs = (
+      ClippingParams(0, None, None, layerRgbClipping.redMin, layerRgbClipping.redMax)
+        :: ClippingParams(1, None, None, layerRgbClipping.greenMin, layerRgbClipping.greenMax)
+        :: ClippingParams(2, None, None, layerRgbClipping.blueMin, layerRgbClipping.blueMax)
+        :: Nil
+    )
+
+    val normalizeLayer =
+      (_:MultibandTile).mapBands { (i, tile) =>
+        val maybeNewTile = for {
+          args <- layerNormalizeArgs.find(cp => cp.band == i)
+          nmax <- rgbBand(args.max, params.max, args.layerMax)
+          nmin <- rgbBand(args.min, params.min, args.layerMin)
+        } yield {
+          normalizeAndClamp(tile, oldMin = nmin, oldMax = nmax, newMin = 0, newMax = 255).convert(UByteConstantNoDataCellType)
+        }
+
+        maybeNewTile.getOrElse(tile)
+      }
     
-    case class ClippingParams(band: Int, min: Option[Int], max: Option[Int])
-    val rgbBandArgs = (
-      ClippingParams(0, params.redMin, params.redMax)
-        :: ClippingParams(1, params.greenMin, params.greenMax)
-        :: ClippingParams(2, params.blueMin, params.blueMax)
+    val colorCorrectArgs = (
+      ClippingParams(0, params.redMin, params.redMax, 0, 255)
+        :: ClippingParams(1, params.greenMin, params.greenMax, 0, 255)
+        :: ClippingParams(2, params.blueMin, params.blueMax, 0, 255) 
         :: Nil
     )
 
-    val normalizeAndClampValues =
+    val applyColorCorrections =
       (_:MultibandTile).mapBands { (i, tile) =>
         val maybeNewTile = for {
-          args <- rgbBandArgs.find(cp => cp.band == i)
-          nmax <- rgbBand(args.max, params.max, maxCellValue(tile.cellType))
-          nmin <- rgbBand(args.min, params.min, 0)
-        } yield normalizeAndClamp(tile, oldMin = nmin, oldMax = nmax, newMin = 0, newMax = 255).convert(UByteConstantNoDataCellType)
+          args <- colorCorrectArgs.find(cp => cp.band == i)
```

Comment:
It seems like this usage is a more natural fit for a map than a list (same comment for Line 189 as well).